src/Symfony/Component/Mailer/Bridge/Sendgrid/Transport/SendgridSmtpTransport.php

```diff
@@ -27,4 +32,45 @@ public function __construct(#[\SensitiveParameter] string $key, ?EventDispatcher
         $this->setUsername('apikey');
         $this->setPassword($key);
     }
+
+    public function send(RawMessage $message, ?Envelope $envelope = null): ?SentMessage
+    {
+        if ($message instanceof Message) {
+            $this->addSendgridHeaders($message);
+        }
+
+        return parent::send($message, $envelope);
+    }
+
+    private function addSendgridHeaders(Message $message): void
+    {
+        $headers = $message->getHeaders();
+
+        if (!$headers->has('X-SMTPAPI')) {
```

Comment:
let's use early return statements (same below)
```suggestion
        if ($headers->has('X-SMTPAPI')) {
            return;
```

Reply:
Code review edits made as part of 514ce3b